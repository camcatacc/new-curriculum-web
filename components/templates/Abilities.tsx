// Modules
import React from "react";

// Elements
import SectionTitle from "components/molecules/SectionTitle";
import GroupAbilities from "components/organisms/GroupAbilities";

// Styles
import { darkPink, brightPink, white } from "styles/customColors";

// Definitions
import type { AbilityProps } from "components/molecules/Ability";
import type { UnformattedBarAbilityProps, FormatBarAbilityProps } from "components/molecules/BarAbility";

export interface AbilitiesProps {
	title: string;
	abilities: AbilityProps[];
}

// Element
const Abilities = ({ title, abilities }: AbilitiesProps) => {
	const barAbilitiesArray: UnformattedBarAbilityProps[] = [
		{
			title: "HTML",
			percentage: 80
		},
		{
			title: "CSS",
			percentage: 80
		},
		{
			title: "Javascript",
			percentage: 90
		},
		{
			title: "Typescript",
			percentage: 90
		},
		{
			title: "React",
			percentage: 90
		},
		{
			title: "Angular",
			percentage: 50
		}
	];

	const barAbilitiesFormat: FormatBarAbilityProps = {
		barStyle: {
			backgroundColor: darkPink
		},
		titleStyle: {
			backgroundColor: brightPink,
			fontColor: white
		},
		fontSize: "1.1rem"
	};

	return (
		<div className="md:min-h-screen">
			<SectionTitle>{title}</SectionTitle>
			<GroupAbilities abilities={abilities} />
			{/*       <div className="flex flex-row gap-32">
        <GroupBarHabilities
          abilities={barAbilitiesArray}
          {...barAbilitiesFormat}
        />
        <GroupBarHabilities
          abilities={barAbilitiesArray}
          {...barAbilitiesFormat}
        />
      </div>
 */}
		</div>
	);
};

export default Abilities;
