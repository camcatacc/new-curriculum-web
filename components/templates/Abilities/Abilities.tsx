// Modules
import React from "react";

// Elements
import SectionTitle from "components/molecules/SectionTitle/SectionTitle";
import GroupAbilities from "components/organisms/GroupAbilities/GroupAbilities";
import GroupBarAbilities from "components/organisms/GroupBarAbilities/GroupBarAbilities";

// Definitions
import type { CmsGroupBarAbilitiesList, CmsAbilitiesList } from "interfaces/cms/pages/Abilities";

export interface AbilitiesProps {
	title: string;
	abilitiesLists: (CmsGroupBarAbilitiesList | CmsAbilitiesList)[];
}

// Element
const Abilities = ({ title, abilitiesLists }: AbilitiesProps) => (
	<div className="md:min-h-screen">
		<SectionTitle>{title}</SectionTitle>
		<div className="flex flex-1 flex-col gap-20">
			{abilitiesLists.map((abList) => {
				switch (abList.type) {
					case "AbilitiesList":
						return <GroupAbilities abilities={abList.abilities.map((ab) => ab.fields)} />;
					case "GroupBarAbilitiesList":
						return (
							<div className="flex flex-row gap-x-32 gap-y-10 flex-wrap justify-center">
								{abList.abilitiesGroups.map((abGroup) => (
									<GroupBarAbilities {...abGroup.fields} abilities={abGroup.fields.abilities.map((ab) => ab.fields)} />
								))}
							</div>
						);
				}
			})}
		</div>
	</div>
);

export default Abilities;
