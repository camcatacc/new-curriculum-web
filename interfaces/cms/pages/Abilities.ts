// Constants
export const CONTENT_TYPE_PRESENTATION_PAGE = "abilitiesPage";

// Definitions
import type { ContentfulEntry } from "interfaces/cms/contentful";
import type { CmsAbility } from "interfaces/cms/molecules/Ability";
import type { CmsGroupBarAbilities } from "interfaces/cms/organisms/GroupBarAbilities/GroupBarAbilities";

export interface CmsAbilitiesPage {
	title: string;
	abilitiesLists: ContentfulEntry<CmsAbilitiesList | CmsGroupBarAbilitiesList>[];
}

export interface CmsAbilitiesList {
	title: string;
	type: "AbilitiesList";
	abilities: ContentfulEntry<CmsAbility>[];
}

export interface CmsGroupBarAbilitiesList {
	title: string;
	type: "GroupBarAbilitiesList";
	abilitiesGroups: ContentfulEntry<CmsGroupBarAbilities>[];
}
