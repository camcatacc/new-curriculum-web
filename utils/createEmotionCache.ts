// Modules
import createCache from "@emotion/cache";

// Functions
export default function createEmotionCache() {
	return createCache({ key: "css" });
}
