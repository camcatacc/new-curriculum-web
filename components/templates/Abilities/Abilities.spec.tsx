// Modules
import React from "react";
import { BLOCKS } from "@contentful/rich-text-types";

// Test Modules
import { customRender } from "utils/tests";

// Accessibility Tests
import { axe, toHaveNoViolations } from "jest-axe";

// Components
import Abilities, { AbilitiesProps } from "components/templates/Abilities/Abilities";

// Accessibility extension
expect.extend(toHaveNoViolations);

// Default Components
const defaultProps: AbilitiesProps = {
	abilitiesLists: [
		{
			abilities: [
				{
					metadata: {
						tags: []
					},
					sys: {
						space: {
							sys: {
								type: "Link",
								linkType: "Space",
								id: "or8z0a5f48r5"
							}
						},
						id: "2wDwDei4PJ7P5Op07sWj4d",
						type: "Entry",
						createdAt: "2021-11-08T16:55:45.542Z",
						updatedAt: "2021-11-11T09:03:51.947Z",
						environment: {
							sys: {
								id: "master",
								type: "Link",
								linkType: "Environment"
							}
						},
						revision: 3,
						contentType: {
							sys: {
								type: "Link",
								linkType: "ContentType",
								id: "ability"
							}
						},
						locale: "en-US"
					},
					fields: {
						title: "Perfectionist",
						text: {
							data: {},
							content: [
								{
									data: {},
									content: [
										{
											data: {},
											marks: [],
											value: "From responsive designs to error absence, nothing is irrelevant.",
											nodeType: "text"
										}
									],
									nodeType: BLOCKS.PARAGRAPH
								}
							],
							nodeType: BLOCKS.DOCUMENT
						},
						icon: "M18 2C17.1 2 16 3 16 4H8C8 3 6.9 2 6 2H2V11C2 12 3 13 4 13H6.2C6.6 15 7.9 16.7 11 17V19.08C8 19.54 8 22 8 22H16C16 22 16 19.54 13 19.08V17C16.1 16.7 17.4 15 17.8 13H20C21 13 22 12 22 11V2H18M6 11H4V4H6V11M20 11H18V4H20V11Z"
					}
				},
				{
					metadata: {
						tags: []
					},
					sys: {
						space: {
							sys: {
								type: "Link",
								linkType: "Space",
								id: "or8z0a5f48r5"
							}
						},
						id: "6ObHPjfR0K6hlcC7XGwqun",
						type: "Entry",
						createdAt: "2021-11-08T16:54:33.124Z",
						updatedAt: "2021-11-11T09:05:17.748Z",
						environment: {
							sys: {
								id: "master",
								type: "Link",
								linkType: "Environment"
							}
						},
						revision: 2,
						contentType: {
							sys: {
								type: "Link",
								linkType: "ContentType",
								id: "ability"
							}
						},
						locale: "en-US"
					},
					fields: {
						title: "Agile Team Player",
						text: {
							data: {},
							content: [
								{
									data: {},
									content: [
										{
											data: {},
											marks: [],
											value: "I know and apply ",
											nodeType: "text"
										},
										{
											data: {},
											marks: [
												{
													type: "bold"
												}
											],
											value: "Agile methodologies",
											nodeType: "text"
										},
										{
											data: {},
											marks: [],
											value: " and consider teamworking of utmost importance.",
											nodeType: "text"
										}
									],
									nodeType: BLOCKS.PARAGRAPH
								}
							],
							nodeType: BLOCKS.DOCUMENT
						},
						icon: "M11 6H14L17.29 2.7A1 1 0 0 1 18.71 2.7L21.29 5.29A1 1 0 0 1 21.29 6.7L19 9H11V11A1 1 0 0 1 10 12A1 1 0 0 1 9 11V8A2 2 0 0 1 11 6M5 11V15L2.71 17.29A1 1 0 0 0 2.71 18.7L5.29 21.29A1 1 0 0 0 6.71 21.29L11 17H15A1 1 0 0 0 16 16V15H17A1 1 0 0 0 18 14V13H19A1 1 0 0 0 20 12V11H13V12A2 2 0 0 1 11 14H9A2 2 0 0 1 7 12V9Z"
					}
				}
			],
			title: "Main Abilities",
			type: "AbilitiesList"
		},
		{
			title: "Main Bar Abilities",
			type: "GroupBarAbilitiesList",
			abilitiesGroups: [
				{
					metadata: {
						tags: []
					},
					sys: {
						space: {
							sys: {
								type: "Link",
								linkType: "Space",
								id: "or8z0a5f48r5"
							}
						},
						id: "1b0UZLfybdSOMege5FmMUc",
						type: "Entry",
						createdAt: "2021-11-11T10:21:46.014Z",
						updatedAt: "2021-11-11T10:29:13.447Z",
						environment: {
							sys: {
								id: "master",
								type: "Link",
								linkType: "Environment"
							}
						},
						revision: 3,
						contentType: {
							sys: {
								type: "Link",
								linkType: "ContentType",
								id: "groupBarAbilities"
							}
						},
						locale: "en-US"
					},
					fields: {
						title: "Web",
						abilities: [
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "1prQS4NyURnS3j8nvyG9Wg",
									type: "Entry",
									createdAt: "2021-11-11T10:27:33.656Z",
									updatedAt: "2021-11-11T10:27:33.656Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "HTML",
									percentage: 80
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "RA9KLKu6wk8pVPnMpZ9OM",
									type: "Entry",
									createdAt: "2021-11-11T10:27:54.316Z",
									updatedAt: "2021-11-11T10:27:54.316Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "CSS",
									percentage: 80
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "5HcwsqorvoWEokKMuwZeH7",
									type: "Entry",
									createdAt: "2021-11-11T10:28:16.211Z",
									updatedAt: "2021-11-11T10:28:16.211Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "JavaScript",
									percentage: 90
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "6z3VBKdxFfhxnFTFVfbx18",
									type: "Entry",
									createdAt: "2021-11-11T10:28:32.546Z",
									updatedAt: "2021-11-11T10:28:32.546Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "TypeScript",
									percentage: 90
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "xr5X272NShacIVBBd33YN",
									type: "Entry",
									createdAt: "2021-11-11T10:28:46.102Z",
									updatedAt: "2021-11-11T10:28:46.102Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "React",
									percentage: 90
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "1hPM9hF78cX5kmOvQusAuF",
									type: "Entry",
									createdAt: "2021-11-11T10:28:57.886Z",
									updatedAt: "2021-11-11T10:28:57.886Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "Angular",
									percentage: 30
								}
							}
						],
						titleColor: "#E25487",
						barColor: "#B2436B"
					}
				},
				{
					metadata: {
						tags: []
					},
					sys: {
						space: {
							sys: {
								type: "Link",
								linkType: "Space",
								id: "or8z0a5f48r5"
							}
						},
						id: "2Pr95TGtfllGTQ62uqOwUX",
						type: "Entry",
						createdAt: "2021-11-11T10:32:09.060Z",
						updatedAt: "2021-11-11T10:32:09.060Z",
						environment: {
							sys: {
								id: "master",
								type: "Link",
								linkType: "Environment"
							}
						},
						revision: 1,
						contentType: {
							sys: {
								type: "Link",
								linkType: "ContentType",
								id: "groupBarAbilities"
							}
						},
						locale: "en-US"
					},
					fields: {
						title: "Mobile",
						abilities: [
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "1jE6B4GvzIkg3r3HeavClM",
									type: "Entry",
									createdAt: "2021-11-11T10:30:32.538Z",
									updatedAt: "2021-11-11T10:30:32.538Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "React Native",
									percentage: 90
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "4soF5A1IaF8SmxKgbKlmbn",
									type: "Entry",
									createdAt: "2021-11-11T10:30:45.207Z",
									updatedAt: "2021-11-11T10:30:45.207Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "Flutter",
									percentage: 70
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "73kK6Yv6Kr29PiF1jgxXUQ",
									type: "Entry",
									createdAt: "2021-11-11T10:31:00.082Z",
									updatedAt: "2021-11-11T10:31:00.082Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "Ionic",
									percentage: 60
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "2mMd4hW1VTAThwaSzLEkXg",
									type: "Entry",
									createdAt: "2021-11-11T10:31:28.373Z",
									updatedAt: "2021-11-11T10:31:28.373Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "Android Dev.",
									percentage: 50
								}
							},
							{
								metadata: {
									tags: []
								},
								sys: {
									space: {
										sys: {
											type: "Link",
											linkType: "Space",
											id: "or8z0a5f48r5"
										}
									},
									id: "7inyS7pHOL00hBrOduSyGx",
									type: "Entry",
									createdAt: "2021-11-11T10:31:42.759Z",
									updatedAt: "2021-11-11T10:31:42.759Z",
									environment: {
										sys: {
											id: "master",
											type: "Link",
											linkType: "Environment"
										}
									},
									revision: 1,
									contentType: {
										sys: {
											type: "Link",
											linkType: "ContentType",
											id: "barAbility"
										}
									},
									locale: "en-US"
								},
								fields: {
									name: "iOS Dev.",
									percentage: 50
								}
							}
						],
						titleColor: "#70C541",
						barColor: "#60A13C"
					}
				}
			]
		}
	],
	title: "Abilities"
};

// Custom rendering
function renderComponent(newProps?: object) {
	const props = { ...defaultProps, ...newProps };
	return customRender(<Abilities {...props} />);
}

describe("Abilities", () => {
	it("Snapshot and accessibility tests", async () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
