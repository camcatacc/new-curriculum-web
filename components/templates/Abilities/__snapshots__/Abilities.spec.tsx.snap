// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abilities Snapshot and accessibility tests 1`] = `
<div>
  <div
    class="md:min-h-screen"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          Abilities
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div
      class="flex flex-1 flex-col md:gap-20 gap-14"
    >
      <div
        class="grid md:gap-x-12 gap-y-10 grid-cols-2 md:grid-cols-3 gap-x-4"
      >
        <div
          class="w-full h-full flex-col flex items-center"
        >
          <div
            class="md:w-1/5 w-2/5"
          >
            <div
              class="sc-bdvvtL gnQPDc rounded-full relative flex"
              data-testid="container"
              height="100%"
              width="100%"
            >
              <div
                class="h-full flex justify-center items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root sc-gsDKAQ fRMNeN"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    color="#3CEB93"
                    d="M18 2C17.1 2 16 3 16 4H8C8 3 6.9 2 6 2H2V11C2 12 3 13 4 13H6.2C6.6 15 7.9 16.7 11 17V19.08C8 19.54 8 22 8 22H16C16 22 16 19.54 13 19.08V17C16.1 16.7 17.4 15 17.8 13H20C21 13 22 12 22 11V2H18M6 11H4V4H6V11M20 11H18V4H20V11Z"
                  />
                </svg>
              </div>
              <div
                style="margin-top: 100%;"
              />
            </div>
          </div>
          <div
            class="mt-4 mb-2"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
            >
              Perfectionist
            </h3>
          </div>
          <div>
            <p
              class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body2 MuiTypography-alignCenter"
            >
              From responsive designs to error absence, nothing is irrelevant.
            </p>
          </div>
        </div>
        <div
          class="w-full h-full flex-col flex items-center"
        >
          <div
            class="md:w-1/5 w-2/5"
          >
            <div
              class="sc-bdvvtL gnQPDc rounded-full relative flex"
              data-testid="container"
              height="100%"
              width="100%"
            >
              <div
                class="h-full flex justify-center items-center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root sc-gsDKAQ fRMNeN"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    color="#3CEB93"
                    d="M11 6H14L17.29 2.7A1 1 0 0 1 18.71 2.7L21.29 5.29A1 1 0 0 1 21.29 6.7L19 9H11V11A1 1 0 0 1 10 12A1 1 0 0 1 9 11V8A2 2 0 0 1 11 6M5 11V15L2.71 17.29A1 1 0 0 0 2.71 18.7L5.29 21.29A1 1 0 0 0 6.71 21.29L11 17H15A1 1 0 0 0 16 16V15H17A1 1 0 0 0 18 14V13H19A1 1 0 0 0 20 12V11H13V12A2 2 0 0 1 11 14H9A2 2 0 0 1 7 12V9Z"
                  />
                </svg>
              </div>
              <div
                style="margin-top: 100%;"
              />
            </div>
          </div>
          <div
            class="mt-4 mb-2"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
            >
              Agile Team Player
            </h3>
          </div>
          <div>
            <p
              class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body2 MuiTypography-alignCenter"
            >
              I know and apply 
              <b>
                Agile methodologies
              </b>
               and consider teamworking of utmost importance.
            </p>
          </div>
        </div>
      </div>
      <div
        class="flex flex-row gap-x-32 gap-y-10 flex-wrap justify-center"
      >
        <div
          class="flex-1"
        >
          <div
            class="mb-5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
            >
              Web
            </h3>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    HTML
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye bsCiNx pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        80%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    CSS
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye bsCiNx pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        80%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    JavaScript
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye bQgcro pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        90%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    TypeScript
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye bQgcro pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        90%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    React
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye bQgcro pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        90%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg lobqhD pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    Angular
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye rZHVm pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        30%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex-1"
        >
          <div
            class="mb-5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
            >
              Mobile
            </h3>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg efprcS pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    React Native
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye fJmpVR pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        90%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg efprcS pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    Flutter
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye ivjyLL pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        70%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg efprcS pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    Ionic
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye fLoBNS pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        60%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg efprcS pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    Android Dev.
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye glwMMt pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        50%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-row justify-center"
            >
              <div
                class="w-40 md:w-32 truncate"
              >
                <div
                  class="sc-jRQBWg efprcS pl-2 pr-2 pt-1 pb-1"
                >
                  <p
                    class="MuiTypography-root sc-gKclnd flksqq w-full truncate MuiTypography-body2 MuiTypography-alignCenter"
                    font-size="1rem"
                    fontcolor="white"
                  >
                    iOS Dev.
                  </p>
                </div>
              </div>
              <div
                class="w-full md:w-80 "
              >
                <div
                  class="bg-gray-300 h-full"
                >
                  <div
                    class="sc-hKwDye glwMMt pr-4 h-full flex flex-col justify-center"
                  >
                    <div>
                      <p
                        class="MuiTypography-root sc-eCImPb DUHjf MuiTypography-body1 MuiTypography-alignRight"
                        font-size="1rem"
                        fontcolor="white"
                      >
                        50%
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
